samples/materials/gltf2.mat

```diff
@@ -0,0 +1,86 @@
+material {
+    name : Lit,
+    requires : [
+        uv0
+    ],
+    shadingModel : lit,
+    parameters : [
+        {
+            type : sampler2d,
+            name : baseColorMap
+        },
+        {
+            type : float4,
+            name : baseColorFactor
+        },
+        {
+            type : sampler2d,
+            name : metallicRoughnessMap
+        },
+        {
+            type : float,
+            name : metallicFactor
+        },
+        {
+            type : float,
+            name : roughnessFactor
+        },
+        {
+            type : sampler2d,
+            name : normalMap
+        },
+        {
+            type : float,
+            name : normalScale
+        },
+        {
+            type : sampler2d,
+            name : aoMap
+        },
+        {
+            type : float,
+            name : aoStrength
+        },
+        {
+            type : sampler2d,
+            name : emissiveMap
+        },
+        {
+            type : float3,
+            name : emissiveFactor
+        }
+    ],
+}
+
+fragment {
+//    void material(inout MaterialInputs material) {
+//        material.normal = texture(materialParams_normalMap, getUV0()).xyz * 2.0 - 1.0;
+//        prepareMaterial(material);
+//        material.baseColor = texture(materialParams_baseColorMap, getUV0());
+//        material.roughness = texture(materialParams_metallicRoughnessMap, getUV0()).g;
+//        material.metallic = texture(materialParams_metallicRoughnessMap, getUV0()).b;
+//        material.ambientOcclusion = texture(materialParams_aoMap, getUV0()).r;
+//    }
+
+void material(inout MaterialInputs material) {
+    material.normal = texture(materialParams_normalMap, getUV0()).xyz * 2.0 - 1.0;
+    material.normal.y = -material.normal.y;
+    prepareMaterial(material);
+//    material.baseColor.rgb = float3(1.0, 0.75, 0.94);
+//    material.metallic = 0.0;
+//    material.roughness = 1.0;
+//    material.ambientOcclusion = 1.0;
+    material.baseColor = texture(materialParams_baseColorMap, getUV0());
```

Comment:
You can simplify this (and other similar cases) to
```
material.baseColor.rgb *= materialParams.baseColorFactor.xyz;
```

Reply:
Also remember that Filament expects premultiplied alpha. So you need:

`material.baseColor.rgb *= material.baseColor.a;`